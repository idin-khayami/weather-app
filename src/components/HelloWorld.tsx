import React from 'react';

const HelloWorld: React.FC = () => {
  return <div>Hello Here</div>;
};

export default HelloWorld;
