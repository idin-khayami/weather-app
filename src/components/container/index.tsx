import { Container } from '@/components/container/Container';

export { Container };
