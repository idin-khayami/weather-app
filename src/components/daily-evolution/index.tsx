import { DailyEvolution } from '@/components/daily-evolution/DailyEvolution';

export { DailyEvolution };
