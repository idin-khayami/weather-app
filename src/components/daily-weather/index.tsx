import { DailyWeather } from '@/components/daily-weather/DailyWeather';

export { DailyWeather };
