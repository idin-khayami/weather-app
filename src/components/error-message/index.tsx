import { ErrorMessage } from '@/components/error-message/ErrorMessage';

export { ErrorMessage };
