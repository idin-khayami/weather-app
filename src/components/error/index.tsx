import { Error } from '@/components/error/Error';

export { Error };
