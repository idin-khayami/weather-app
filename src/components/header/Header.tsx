import clsx from 'clsx';
import Link from 'next/link';

import { ComponentBaseProps } from '@/types/global.types';
import { paths } from '@/utils/paths';

import styles from './header.module.scss';

interface HeaderProps extends ComponentBaseProps {}

export const Header = ({ dataTestId, className }: HeaderProps) => {
  return (
    <header
      data-testid={dataTestId}
      className={clsx([styles.header, className])}
    >
      <Link href={paths.homepage}>
        <p className={styles.title}>React Weather</p>
      </Link>
    </header>
  );
};
