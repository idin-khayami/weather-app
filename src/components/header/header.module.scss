@import '../../styles/vars';

.header {
  width: 100%;
  height: 95px;
  background-color: $headerBackground;
  padding: 14px 24px;
  @include flex(row, space-between, center);
}

.title {
  color: $whiteColor; 
  font-size: 36px;
  font-weight: 900;
  @include flex(row, flex-start, center);
}
