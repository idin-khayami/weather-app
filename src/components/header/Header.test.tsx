import { render, screen } from '@testing-library/react';

import { Header } from '@/components/header';

describe('Component: header component', () => {
  it('should match the snapshot', () => {
    render(<Header dataTestId="testingHeader" />);

    const header = screen.getByTestId('testingHeader');

    expect(header).toMatchSnapshot();
  });
});
