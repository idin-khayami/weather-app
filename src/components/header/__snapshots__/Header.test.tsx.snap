// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: header component should match the snapshot 1`] = `
<header
  class="header"
  data-testid="testingHeader"
>
  <a
    href="/"
  >
    <p
      class="title"
    >
      React Weather
    </p>
  </a>
</header>
`;
